Microsoft.AspNet.Mvc.Razor.CompilationFailedException when hosting a simple MVC app on IIS

1. New app from Empty template, add a simple controller with a view
   -- Repro project is available here 
2. Select ActiveTargetFramework to core CLR
3. F5 works
4. Publish to File System

```
"C:\Users\balach\.kre\packages\kre-svr50-x86.0.1-alpha-build-0443\bin\kpm.cmd" "pack" --runtime "C:\Users\balach\.kre\packages\kre-svrc50-x86.0.1-alpha-build-0443" --out "C:\PublishOut\EmptyApp1\P4" --overwrite
```
1. Map the output folder to a new website on IIS, copy the x64 packages and browse the site on IIS
## ACTUAL

On IIS
Error when we browse the site on IIS

```
:(
500 Internal Server Error 
Microsoft.AspNet.Mvc.Razor.CompilationFailedException: Compilation Failed:(11,81): error CS0246: The type or namespace name 'User1' could not be found (are you missing a using directive or an assembly reference?)
at Microsoft.AspNet.Mvc.Razor.CompilationResult.get_CompiledType()
at Microsoft.AspNet.Mvc.Razor.CompilerCache.GetOrAdd(IFileInfo file, Func`1 compile)
at Microsoft.AspNet.Mvc.Razor.RazorCompilationService.Compile(IFileInfo file)
at Microsoft.AspNet.Mvc.Razor.VirtualPathViewFactory.CreateInstance(String virtualPath)
at Microsoft.AspNet.Mvc.Razor.RazorViewEngine.CreateViewEngineResult(IDictionary`2 context, String viewName)
at Microsoft.AspNet.Mvc.Razor.RazorViewEngine.FindView(IDictionary`2 context, String viewName)
at Microsoft.AspNet.Mvc.ViewResult.FindView(IDictionary`2 context, String viewName)
at Microsoft.AspNet.Mvc.ViewResult.<ExecuteResultAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionResult>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionResultFilter>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionResultWithFilters>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Mvc.ReflectedActionInvoker.<InvokeActionAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Mvc.MvcRouteHandler.<RouteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Routing.Template.TemplateRoute.<RouteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Routing.RouteCollection.<RouteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Builder.RouterMiddleware.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.RequestContainer.ContainerMiddleware.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Loader.IIS.KlrHttpApplication.<ProcessRequestAsyncImpl>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
at Microsoft.AspNet.Loader.IIS.HttpApplicationBase.<InvokeProcessRequestAsyncImpl>d__1.MoveNext()
```

--- a/src/Microsoft.AspNet.Mvc.Razor.Host/IMvcRazorHost.cs
+++ b/src/Microsoft.AspNet.Mvc.Razor.Host/IMvcRazorHost.cs
@@ -8,6 +8,6 @@ namespace Microsoft.AspNet.Mvc.Razor
 {
     public interface IMvcRazorHost
     {
-        GeneratorResults GenerateCode(string rootNamespace, string rootRelativePath, Stream inputStream);
+        GeneratorResults GenerateCode(string rootRelativePath, Stream inputStream);
     }
 }

--- a/src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs
+++ b/src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs
@@ -11,6 +11,8 @@ namespace Microsoft.AspNet.Mvc.Razor
 {
     public class MvcRazorHost : RazorEngineHost, IMvcRazorHost
     {
+        private const string ViewNamespace = "ASP";
+
         private static readonly string[] _defaultNamespaces = new[] 
         { 
             "System",
@@ -53,13 +55,13 @@ namespace Microsoft.AspNet.Mvc.Razor
             }
         }
 
-        public GeneratorResults GenerateCode(string rootNamespace, string rootRelativePath, Stream inputStream)
+        public GeneratorResults GenerateCode(string rootRelativePath, Stream inputStream)
         {
             string className = ParserHelpers.SanitizeClassName(rootRelativePath);
             using (var reader = new StreamReader(inputStream))
             {
                 var engine = new RazorTemplateEngine(this);
-                return engine.GenerateCode(reader, className, rootNamespace, rootRelativePath);
+                return engine.GenerateCode(reader, className, ViewNamespace, rootRelativePath);
             }
         }
 

--- a/src/Microsoft.AspNet.Mvc.Razor/Razor/RazorCompilationService.cs
+++ b/src/Microsoft.AspNet.Mvc.Razor/Razor/RazorCompilationService.cs
@@ -42,7 +42,7 @@ namespace Microsoft.AspNet.Mvc.Razor
             {
                 Contract.Assert(file.PhysicalPath.StartsWith(_appRoot, StringComparison.OrdinalIgnoreCase));
                 var rootRelativePath = file.PhysicalPath.Substring(_appRoot.Length);
-                results = _razorHost.GenerateCode(_environment.ApplicationName, rootRelativePath, inputStream);
+                results = _razorHost.GenerateCode(rootRelativePath, inputStream);
             }
 
             if (!results.Success)

--- a/test/Microsoft.AspNet.Mvc.Razor.Test/RazorCompilationServiceTest.cs
+++ b/test/Microsoft.AspNet.Mvc.Razor.Test/RazorCompilationServiceTest.cs
@@ -24,7 +24,7 @@ namespace Microsoft.AspNet.Mvc.Razor.Test
             env.SetupGet(e => e.ApplicationName).Returns("MyTestApplication");
             env.SetupGet(e => e.ApplicationBasePath).Returns(appPath);
             var host = new Mock<IMvcRazorHost>();
-            host.Setup(h => h.GenerateCode("MyTestApplication", @"views\index\home.cshtml", It.IsAny<Stream>()))
+            host.Setup(h => h.GenerateCode(@"views\index\home.cshtml", It.IsAny<Stream>()))
                 .Returns(new GeneratorResults(new Block(new BlockBuilder { Type = BlockType.Comment }), new RazorError[0], new CodeBuilderResult("", new LineMapping[0])))
                 .Verifiable();
             var compiler = new Mock<ICompilationService>();

